readthedocs/doc_builder/environments.py

```diff
@@ -726,10 +727,18 @@ def __init__(self, *args, **kwargs):
                 project_name=self.project.slug,
             )[:DOCKER_HOSTNAME_MAX_LEN],
         )
+
+        # Decide what Docker image to use, based on priorities:
+        # Use the Docker image set by user or,
         if self.config and self.config.build.image:
             self.container_image = self.config.build.image
+        # the image overridden by the project (manually set by an admin) or,
         if self.project.container_image:
             self.container_image = self.project.container_image
```

Comment:
I also wonder if this should be before the users config? I can see reasons for both, but having our DB entry overwrite the users wishes seems really confusing.

Reply:
We already to this from the config module

https://github.com/rtfd/readthedocs.org/blob/7aa57bbcd01994b662f34e3a7bf7ea8cb7ab82ac/readthedocs/doc_builder/config.py#L55

https://github.com/rtfd/readthedocs.org/blob/7aa57bbcd01994b662f34e3a7bf7ea8cb7ab82ac/readthedocs/config/config.py#L415-L419

https://github.com/rtfd/readthedocs.org/blob/7aa57bbcd01994b662f34e3a7bf7ea8cb7ab82ac/readthedocs/config/config.py#L703-L707

Reply:
Hrm ok. I guess the DB setting makes a bit of sense to overwrite, but not a feature flag for testing.